XcodeServerSDK/Server Entities/Issue.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Issue.swift
+//  XcodeServerSDK
+//
+//  Created by Mateusz Zając on 04.08.2015.
+//  Copyright © 2015 Honza Dvorsky. All rights reserved.
+//
+
+import Foundation
+
+public class Issue: XcodeServerEntity {
+    
+    public enum Type: String {
+        case BuildServiceError = "buildServiceError"
+        // Looks like build service warnings are treated as warnings in case
+        // of type. In case of Issue Type they're Build Service Error
+        // case BuildServiceWarining = "warning"
+        case TriggerError = "triggerError"
+        case Error = "error"
+        case Warning = "warning"
+        case TestFailure = "testFailure"
+        case AnalyzerWarning = "analyzerWarning"
```

Comment:
I must be missing something. Below I made sure I had a normal warning and a build service warning both in the same integration, and the `type` is `warning` and `buildServiceWarning`. What do you mean you can't tell them apart? I think we need another value in this enum: `case BuildServiceWarning = "buildServiceWarning"`.

```
{
    "analyzerWarnings": {
        "freshIssues": [], 
        "resolvedIssues": [], 
        "silencedIssues": [], 
        "unresolvedIssues": []
    }, 
    "buildServiceErrors": [], 
    "buildServiceWarnings": [
        {
            "_id": "97d1a7bff431c8f005a2623d4c09d88a", 
            "_rev": "3-763793ea504595b4b806b553f36a1fbf", 
            "age": 0, 
            "commits": [], 
            "integrationID": "97d1a7bff431c8f005a2623d4c093b90", 
            "issueType": "buildServiceWarning", 
            "message": "code signing is required for product type 'Application' in SDK 'iOS 9.0'", 
            "status": 0, 
            "type": "buildServiceWarning"
        }, 
        {
            "_id": "97d1a7bff431c8f005a2623d4c09d8e5", 
            "_rev": "3-f596c2a933fa0b660c5b2dd3ec1201e2", 
            "age": 0, 
            "commits": [], 
            "integrationID": "97d1a7bff431c8f005a2623d4c093b90", 
            "issueType": "buildServiceWarning", 
            "message": "Installable product could not be created because creating the archive failed.", 
            "status": 0, 
            "type": "buildServiceWarning"
        }
    ], 
    "errors": {
        "freshIssues": [], 
        "resolvedIssues": [], 
        "silencedIssues": [], 
        "unresolvedIssues": []
    }, 
    "testFailures": {
        "freshIssues": [], 
        "resolvedIssues": [], 
        "silencedIssues": [], 
        "unresolvedIssues": []
    }, 
    "triggerErrors": [], 
    "warnings": {
        "freshIssues": [
            {
                "_id": "97d1a7bff431c8f005a2623d4c09e3fa", 
                "_rev": "3-8634b2c6bf12ca9dd73b8b9b12fb9d65", 
                "age": 0, 
                "commits": [
                    {
                        "XCSBlueprintRepositoryID": "43ABCED2B571DB1C2336479F8D53EBF39F6D1B5D", 
                        "XCSCommitCommitChangeFilePaths": [
                            {
                                "filePath": "XCSTutorialProject1/ViewController.swift", 
                                "status": 4
                            }
                        ], 
                        "XCSCommitContributor": {
                            "XCSContributorDisplayName": "Honza Dvorsky", 
                            "XCSContributorEmails": [
                                "honza@swiftkey.com"
                            ], 
                            "XCSContributorName": "Honza Dvorsky"
                        }, 
                        "XCSCommitHash": "316aec682d7aa1f0cbbba10d46d17cea54373cf9", 
                        "XCSCommitMessage": "intentional warning", 
                        "XCSCommitTimestamp": "2015-08-06T11:00:47.000Z"
                    }
                ], 
                "documentFilePath": "XCSTutorialProject1/XCSTutorialProject1/ViewController.swift", 
                "documentLocationData": "YnBsaXN0MDDUAQIDBAUGIyRYJHZlcnNpb25YJG9iamVjdHNZJGFyY2hpdmVyVCR0b3ASAAGGoKQHCBobVSRudWxs2gkKCwwNDg8QERITFBQVFhcYGRgVW2RvY3VtZW50VVJMXxAUc3RhcnRpbmdDb2x1bW5OdW1iZXJfEBJlbmRpbmdDb2x1bW5OdW1iZXJfEBFjaGFyYWN0ZXJSYW5nZUxlbl8QEGxvY2F0aW9uRW5jb2RpbmdWJGNsYXNzXxAQZW5kaW5nTGluZU51bWJlcll0aW1lc3RhbXBfEBJzdGFydGluZ0xpbmVOdW1iZXJfEBFjaGFyYWN0ZXJSYW5nZUxvY4ACEAwQABABgAMQEIAAXxCeZmlsZTovLy9MaWJyYXJ5L0RldmVsb3Blci9YY29kZVNlcnZlci9JbnRlZ3JhdGlvbnMvQ2FjaGVzLzk3ZDFhN2JmZjQzMWM4ZjAwNWEyNjIzZDRjMDkzNTExL1NvdXJjZS9YQ1NUdXRvcmlhbFByb2plY3QxL1hDU1R1dG9yaWFsUHJvamVjdDEvVmlld0NvbnRyb2xsZXIuc3dpZnTSHB0eH1okY2xhc3NuYW1lWCRjbGFzc2VzXxAXRFZUVGV4dERvY3VtZW50TG9jYXRpb26jICEiXxAXRFZUVGV4dERvY3VtZW50TG9jYXRpb25fEBNEVlREb2N1bWVudExvY2F0aW9uWE5TT2JqZWN0XxAPTlNLZXllZEFyY2hpdmVy0SUmVHJvb3SAAQAIABEAGgAjAC0AMgA3ADwAQgBXAGMAegCPAKMAtgC9ANAA2gDvAQMBBQEHAQkBCwENAQ8BEQGyAbcBwgHLAeUB6QIDAhkCIgI0AjcCPAAAAAAAAAIBAAAAAAAAACcAAAAAAAAAAAAAAAAAAAI+", 
                "integrationID": "97d1a7bff431c8f005a2623d4c093b90", 
                "issueType": "Swift Compiler Warning", 
                "lineNumber": 17, 
                "message": "Initialization of immutable value 'warning' was never used; consider replacing with assignment to '_' or removing it", 
                "status": 0, 
                "target": "XCSTutorialProject1", 
                "type": "warning"
            }
        ], 
        "resolvedIssues": [], 
        "silencedIssues": [], 
        "unresolvedIssues": []
    }
}
```